healthcheck/ticker.go

```diff
@@ -1,35 +1,135 @@
 package healthcheck
 
 import (
-	"github.com/ONSdigital/go-ns/log"
+	"sync"
 	"time"
+
+	"github.com/ONSdigital/go-ns/log"
 )
 
 // Ticker calls the health check monitor function at configured intervals.
 type Ticker struct {
 	timeTicker *time.Ticker
+	asyncCheck chan bool
+	closing    chan bool
+}
+
+// NewTicker returns a new ticker that checks the given clients at intervals of the given duration
+func NewTicker(duration, recoveryDuration time.Duration, clients ...Client) *Ticker {
+	return NewTickerWithAlerts(duration, recoveryDuration, nil, nil, clients...)
 }
 
-// NewTicker returns a new ticker that checks the given clients at intervals of the given duration.
-func NewTicker(duration time.Duration, clients ...Client) *Ticker {
+// NewTickerWithAlerts returns a new ticker that checks the given clients at intervals
+// - intervals vary in length (when healthy, use `duration`; at startup, or unhealthy state, use `recoveryDuration`)
+// - sends alerts of any change in health via the given channel (if non-nil) - with the new state (i.e. `true` if newly-healthy)
+// - healthchecks can be requested using requestCheckChan (sending `false` on this channel indicates that a failure has prompted this check)
```

Comment:
what else would prompt this check? is there a difference between the check thats triggered via false and true?

Reply:
effectively, no difference (just that of the expected _current health state_).
but I'll expand the comment (I imagine fewer reasons for `true` to be used, perhaps in my scenario - kafka-backoffs - but may happen in, say, a web server, where non-green retries are more likely)